### PR TITLE
[CSL-3114] Update docs to reflect new 10k limit

### PR DESCRIPTION
**Pull request by Mudaafi**
Updates the documentation to reflect the new 10k item limit when
- Uploading Items via the Items API
- Uploading Variations via the Variations API
- Uploading Item Groups via the Item Groups API

I didn't regenerate the docs just yet. Will do so before I merge.

### CODE REVIEW

File: src/modules/catalog.js [hunk 1]
```diff
@@ -1119,6 +1119,7 @@ class Catalog {
    *     },
    * });
    */
+
```
Comment (mocca102) [inline src/modules/catalog.js:1122]: extra space here

**Review by mocca102 — APPROVED** [review]
LGTM!

File: src/modules/catalog.js [hunk 2]
```diff
@@ -886,7 +886,7 @@ class Catalog {
   }
 
   /**
-   * Add item groups to index (limit of 1,000)
+   * Add item groups to index (limit of 10,000)
```
[the remark this answers is not in the document]
Comment (Mudaafi) [reply src/modules/catalog.js:889]: that's a good point, let me ask them

[the remark this answers is not in the document]
Comment (Mudaafi) [reply src/modules/catalog.js]: Reverting this to 1k items following backend's lead: https://constructor.slack.com/archives/CA44G41HA/p1709562698499959?thread_ts=1709314392.618379&cid=CA44G41HA

**Review by esezen — APPROVED** [review]
LGTM!

**Pull request merged**